src/lib_json/json_reader.cpp

```diff
@@ -839,14 +837,13 @@ JSONCPP_STRING Reader::getFormattedErrorMessages() const {
 
 std::vector<Reader::StructuredError> Reader::getStructuredErrors() const {
   std::vector<Reader::StructuredError> allErrors;
-  for (Errors::const_iterator itError = errors_.begin();
-       itError != errors_.end(); ++itError) {
-    const ErrorInfo& error = *itError;
-    Reader::StructuredError structured;
+  allErrors.reserve(errors_.size());
```

Comment:
```
std::vector<Reader::StructuredError> allErrors;
allErrors.reserve(errors_.size());
std::transform(errors_.begin(), errors_.end(), std::back_inserter(allErrors),
               [](const ErrorInfo& e) -> Reader::StructuredError {
                 return {e.token_.start_ - begin_,
                         e.token_.end_ - begin_,
                         e.message_};
               });
return allErrors;
```

Reply:
IMHO code without too much std boilerplate might be better
  for (const ErrorInfo& e : errors_) {
    allErrors.push_back(
        { e.token_.start_ - begin_, e.token_.end_ - begin_, e.message_ });
    }